evalml/pipelines/pipeline_base.py

```diff
@@ -194,8 +194,8 @@ def compute_estimator_features(self, X):
         return X_t
 
     def _fit(self, X, y):
-        X_t = X
-        y_t = y
+        X_t = X.reset_index(drop=True)
```

Comment:
Added to avoid a bug when either dataframe/series have indices that aren't aligned with each other